source/audio/Music.cpp

```diff
@@ -217,16 +212,16 @@ void Music::Decode()
 				memcpy(&input.front(), stream.next_frame, remainder);
 
 			// Now, read a chunk of data from the file.
-			size_t read = fread(&input.front() + remainder, 1, INPUT_CHUNK - remainder, file);
+			file->read(reinterpret_cast<char *>(input.data() + remainder), INPUT_CHUNK - remainder);
 			// If you get the end of the file, loop around to the beginning.
-			if(!read || feof(file))
-				rewind(file);
+			if(file->eof())
+				file->seekg(0, iostream::beg);
 			// If there is nothing to decode, return to the top of this loop.
-			if(!(read + remainder))
+			if(!(remainder || !file->eof()))
```

Comment:
```suggestion
			if(!remainder || file->eof())
```
but is the eof check really needed here?

Reply:
Yeah. At the very least, if the mp3 file is empty.

This portion is re-implemented in #11065 anyway.

Reply:
> Also, wouldn't the condition be `!remainder && file->eof()`?

Not sure, but if your current version is correct, then yes.
```suggestion
			if(!remainder && file->eof())
```